decompiler/analysis/analyze_inspect_method.cpp

```diff
@@ -1148,13 +1163,21 @@ std::string TypeInspectorResult::print_as_deftype(
 
   result.append(fmt::format("  :method-count-assert {}\n", type_method_count));
   result.append(fmt::format("  :size-assert         #x{:x}\n", type_size));
+  if (type_heap_base.has_value()) {
+    result.append(fmt::format("  :heap-base           #x{:x}\n", type_heap_base.value()));
```

Comment:
can you check to see if this case is ever hit or not? If not, want to just replace this with an ASSERT?

Reply:
It's hit once:
```lisp
(deftype ocean-wave-frames (structure)
  ((frame ocean-wave-data 64 :offset-assert 0) ;; guessed by decompiler
   )
  :method-count-assert 9
  :size-assert         #x0
  :heap-base           #x1
  :flag-assert         #x900010000
  )
```

Reply:
ah ok. Then leave it as is.

The `ocean-wave-frames` thing is a bug in the original compiler - the size of the type is larger than `UINT16_MAX` but they only use a `u16` to store the size.

I think this happened in jak 1 as well.